pmlc/ast/builder.cc

```diff
@@ -681,16 +681,19 @@ struct ProgramBuilder {
       operands.push_back(builder.lookupNode(operand));
     }
     llvm::SetVector<Value> results;
+    llvm::SetVector<Type> resultTypes;
```

Comment:
I'm not sure this is correct. This SetVector will collapse multiple types into a single entry. But we definitely want to be able to support multiple results with the same type.

Reply:
good point - what I really want is a plain vector, copying the types of the results set.

Reply:
Yes, but be careful, because we *do* want multiple values to be collapsed. So you might need to have a 2nd loop to construct the plain vector.